`wsdl2java` does not work when the module is built from another directory using `mvn -f "module/pom.xml"`

Originally seen in https://github.com/apache/camel-quarkus-examples/actions/runs/5196816411/jobs/9371300458?pr=144

Steps to reproduce:

```
git clone git@github.com:apache/camel-quarkus-examples.git
cd camel-quarkus-examples
mvn clean test -f cxf-soap/ -e
...
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.1.0.Final:generate-code (generate-code) on project camel-quarkus-examples-cxf-soap: Quarkus code generation phase has failed: InvocationTargetException: Could not run wsdl2Java -d target/generated-sources/wsdl2java -b src/main/resources/binding.xml src/main/resources/wsdl/CustomerService.wsdl: java.lang.IllegalArgumentException: URI is not hierarchical -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:3.1.0.Final:generate-code (generate-code) on project camel-quarkus-examples-cxf-soap: Quarkus code generation phase has failed
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:347)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:330)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:175)
        at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:76)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:163)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:160)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
        at io.takari.maven.builder.smart.SmartBuilderImpl.buildProject(SmartBuilderImpl.java:204)
        at io.takari.maven.builder.smart.SmartBuilderImpl$ProjectBuildTask.run(SmartBuilderImpl.java:78)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.maven.plugin.MojoExecutionException: Quarkus code generation phase has failed
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:85)
        at io.quarkus.maven.GenerateCodeMojo.doExecute(GenerateCodeMojo.java:53)
        at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:154)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:342)
        ... 15 common frames omitted
Caused by: java.lang.reflect.InvocationTargetException: null
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:79)
        ... 19 common frames omitted
Caused by: java.lang.RuntimeException: Could not run wsdl2Java -d target/generated-sources/wsdl2java -b src/main/resources/binding.xml src/main/resources/wsdl/CustomerService.wsdl
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.lambda$wsdl2java$1(Wsdl2JavaCodeGen.java:108)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.lambda$scan$2(Wsdl2JavaCodeGen.java:141)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.lambda$scan$3(Wsdl2JavaCodeGen.java:149)
        at io.quarkus.paths.PathTreeVisit.walk(PathTreeVisit.java:25)
        at io.quarkus.paths.DirectoryPathTree.walk(DirectoryPathTree.java:89)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.scan(Wsdl2JavaCodeGen.java:146)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.wsdl2java(Wsdl2JavaCodeGen.java:98)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.trigger(Wsdl2JavaCodeGen.java:78)
        at io.quarkus.deployment.CodeGenerator.lambda$trigger$3(CodeGenerator.java:176)
        at io.quarkus.deployment.CodeGenerator.callWithClassloader(CodeGenerator.java:151)
        at io.quarkus.deployment.CodeGenerator.trigger(CodeGenerator.java:173)
        at io.quarkus.deployment.CodeGenerator.initAndRun(CodeGenerator.java:63)
        ... 24 common frames omitted
Caused by: org.apache.cxf.tools.common.ToolException: java.lang.IllegalArgumentException: URI is not hierarchical
        at org.apache.cxf.tools.wsdlto.WSDLToJavaContainer.execute(WSDLToJavaContainer.java:410)
        at org.apache.cxf.tools.common.toolspec.ToolRunner.runTool(ToolRunner.java:105)
        at org.apache.cxf.tools.wsdlto.WSDLToJava.run(WSDLToJava.java:113)
        at org.apache.cxf.tools.wsdlto.WSDLToJava.run(WSDLToJava.java:86)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.lambda$wsdl2java$1(Wsdl2JavaCodeGen.java:106)
        ... 35 common frames omitted
Caused by: java.lang.IllegalArgumentException: URI is not hierarchical
        at java.base/java.io.File.<init>(File.java:420)
        at org.apache.cxf.resource.URIResolver.tryFileSystem(URIResolver.java:152)
        at org.apache.cxf.resource.URIResolver.<init>(URIResolver.java:93)
        at org.apache.cxf.resource.URIResolver.<init>(URIResolver.java:75)
        at org.apache.cxf.resource.URIResolver.<init>(URIResolver.java:71)
        at org.apache.cxf.tools.wsdlto.frontend.jaxws.JAXWSContainer.validate(JAXWSContainer.java:73)
        at org.apache.cxf.tools.wsdlto.WSDLToJavaContainer.processWsdl(WSDLToJavaContainer.java:164)
        at org.apache.cxf.tools.wsdlto.WSDLToJavaContainer.execute(WSDLToJavaContainer.java:156)
        at org.apache.cxf.tools.wsdlto.WSDLToJavaContainer.execute(WSDLToJavaContainer.java:402)
        ... 39 common frames omitted
```

Apparently, this issue is reproducible only if there is a `-b` option set in `application.properties`:

```
quarkus.cxf.codegen.wsdl2java.additional-params = -b,src/main/resources/binding.xml
```

--- a/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaCodeGen.java
+++ b/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaCodeGen.java
@@ -4,9 +4,11 @@ import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
+import java.util.ListIterator;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
@@ -75,13 +77,21 @@ public class Wsdl2JavaCodeGen implements CodeGenProvider {
         final Wsdl2JavaParameterSet rootParams = buildParameterSet(configFunction, WSDL2JAVA_CONFIG_KEY_PREFIX);
         final Map<String, String> processedFiles = new HashMap<>();
         boolean result = false;
-        result |= wsdl2java(context.inputDir(), rootParams, outDir, WSDL2JAVA_CONFIG_KEY_PREFIX, processedFiles);
+
+        /*
+         * TODO: this is a workaround for https://github.com/quarkusio/quarkus/issues/34422
+         * While context.workDir() returns target or any other direct subdirectory of the project directory
+         * then this workaround will work. But it may fail as long as the project has configured some non-standard
+         * build directory.
+         */
+        final Path projectDir = context.workDir().getParent();
+        result |= wsdl2java(projectDir, context.inputDir(), rootParams, outDir, WSDL2JAVA_CONFIG_KEY_PREFIX, processedFiles);
 
         final Set<String> names = findParamSetNames(config.getPropertyNames());
         for (String name : names) {
             final String prefix = WSDL2JAVA_NAMED_CONFIG_KEY_PREFIX + name;
             final Wsdl2JavaParameterSet namedParams = buildParameterSet(configFunction, prefix);
-            result |= wsdl2java(context.inputDir(), namedParams, outDir, prefix, processedFiles);
+            result |= wsdl2java(projectDir, context.inputDir(), namedParams, outDir, prefix, processedFiles);
         }
 
         if (!result) {
@@ -92,11 +102,13 @@ public class Wsdl2JavaCodeGen implements CodeGenProvider {
         return result;
     }
 
-    static boolean wsdl2java(Path inputDir, Wsdl2JavaParameterSet params, Path outDir, String prefix,
+    static boolean wsdl2java(Path projectDir, Path inputDir, Wsdl2JavaParameterSet params, Path outDir, String prefix,
             Map<String, String> processedFiles) {
 
         return scan(inputDir, params.includes, params.excludes, prefix, processedFiles, (Path wsdlFile) -> {
-            final Wsdl2JavaParams wsdl2JavaParams = new Wsdl2JavaParams(inputDir, outDir, wsdlFile,
+            final Wsdl2JavaParams wsdl2JavaParams = new Wsdl2JavaParams(
+                    projectDir,
+                    inputDir, outDir, wsdlFile,
                     params.additionalParams.orElse(Collections.emptyList()));
             if (log.isInfoEnabled()) {
                 log.info(wsdl2JavaParams.appendLog(new StringBuilder("Running wsdl2java")).toString());
@@ -222,17 +234,36 @@ public class Wsdl2JavaCodeGen implements CodeGenProvider {
     }
 
     static class Wsdl2JavaParams {
+        private final Path projectDir;
         private final Path inputDir;
         private final Path outDir;
         private final Path wsdlFile;
         private final List<String> additionalParams;
 
-        public Wsdl2JavaParams(Path inputDir, Path outDir, Path wsdlFile, List<String> additionalParams) {
+        public Wsdl2JavaParams(Path projectDir, Path inputDir, Path outDir, Path wsdlFile, List<String> additionalParams) {
             super();
+            this.projectDir = projectDir;
             this.inputDir = inputDir;
             this.outDir = outDir;
             this.wsdlFile = wsdlFile;
-            this.additionalParams = additionalParams;
+            this.additionalParams = absolutizeBindings(projectDir, additionalParams);
+        }
+
+        static List<String> absolutizeBindings(Path projectDir, List<String> additionalParams) {
+            List<String> result = new ArrayList<>(additionalParams);
+            ListIterator<String> it = result.listIterator();
+            while (it.hasNext()) {
+                String val = it.next();
+                if ("-b".equals(val) && it.hasNext()) {
+                    String rawBindingFile = it.next();
+                    Path bindingPath = Paths.get(rawBindingFile);
+                    if (!bindingPath.isAbsolute()) {
+                        /* A fix for https://github.com/quarkiverse/quarkus-cxf/issues/907 */
+                        it.set(projectDir.resolve(bindingPath).toString());
+                    }
+                }
+            }
+            return result;
         }
 
         public StringBuilder appendLog(StringBuilder sb) {

